src/main/kotlin/com/hibob/academy/dao/PetDao.kt

```diff
@@ -0,0 +1,79 @@
+
+import com.hibob.academy.dao.*
+import jakarta.inject.Inject
+import org.jooq.DSLContext
+import org.jooq.Record
+import org.jooq.RecordMapper
+import org.springframework.stereotype.Component
+import java.sql.Date
+import kotlin.random.Random
+
+@Component
+
+class PetDao @Inject constructor(private val sql: DSLContext) {
+
+    private val pet = PetTable.instance
+    private val companyId = Random.nextLong()
+
+    private val petMapper = RecordMapper<Record, PetData> { record ->
+        PetData(
+            record[pet.name],
+            convertStringToPetType(record[pet.type]),
+            record[pet.companyId],
+            Date(record[pet.dateOfArrival].time)
+        )
+    }
+
+    private val petMapperWithoutType = RecordMapper<Record, PetDataWithoutType> { record ->
+        PetDataWithoutType(
+            record[pet.name],
+            record[pet.companyId],
+            Date(record[pet.dateOfArrival].time)
+        )
+    }
+
+    fun getPetsByType(petType: PetType): List<PetDataWithoutType> =
+        sql.select(pet.name, pet.companyId, pet.dateOfArrival)
+            .from(pet)
+            .where(pet.type.eq(convertPetTypeToPetString(petType))
+            .and(pet.companyId.eq(companyId)))
+            .fetch(petMapperWithoutType)
+
+    fun getAllPets(): List<PetData> =
+        sql.select(pet.name, pet.type, pet.companyId, pet.dateOfArrival)
+            .from(pet)
+            .where(pet.companyId.eq(companyId))
+            .fetch(petMapper)
+
+
+    fun createPet(name: String, type: String, companyId: Long, dateOfArrival: Date) =
+        sql.insertInto(pet)
+            .set(pet.name, name)
+            .set(pet.type, type)
+            .set(pet.companyId, companyId)
+            .set(pet.dateOfArrival, dateOfArrival)
+            .onConflict(pet.companyId)
+            .doNothing()
+            .execute()
+
+    fun updatePetOwner(petData: PetData, ownerId: Long) =
+        sql.update(pet)
+            .set(pet.ownerId, ownerId)
+            .where(pet.companyId.eq(petData.companyId))
+            .and(pet.ownerId.isNull())
+            .execute()
```

Comment:
In this case you are updating all the pets under some companyId that doesnt have owner.
As mention before, pet should have an Id and you should update only this pet (since id is unique you should use the id in the where).